mathdojo-nonproduction/rules/registerNewUser.js

```diff
@@ -0,0 +1,44 @@
+function registerNewUser(user, context, callback) {
+  // eslint-disable-next-line import/no-unresolved
+  const axios = require("axios@0.19.2");
+  const crypto = require("crypto");
+  const mathDojoNamespace = "http://math-dojo.io/";
+  const orgIdForUser = configuration.userAccountServiceDefaultOrgId;
+  const options = {
+    method: "POST",
+    // eslint-disable-next-line no-undef
+    url: `${configuration.userAccountServiceDomain}/organisations/${orgIdForUser}/users`,
+    headers: {
+      "Content-Type": "application/json",
+      "X-API-Key": `${configuration.userAccountServiceApiKey}`,
+    },
+    data: {
+      name: user.name,
+      profileImageLink: user.picture,
+      accountVerified: user.email_verified,
+      id: crypto
```

Comment:
Sure. The user's id and the provider of their account (Google etc) is sensitive info (i.e. only a third-party, Auth0 or us, can view it with appropriate permissions from them). For that reason, it doesn't seem appropriate to store that in plain text in our database (in the event the undesirable happens, the last thing we want is to have to tell people we leaked sensitive data of theirs). We should probably actually append a secret "salt" to the text before hashing it because that's what would make it truly effective.

Reply:
On reflection, as it's just an id, I don't think a hash would be necessary.